front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -16,6 +24,20 @@ import { apiError, withLogging } from "@app/logger/withlogging";
 
 const { NODE_ENV } = process.env;
 
+const PostManagedDataSourceRequestBodySchema = t.type({
+  provider: t.string,
+  connectionId: t.union([t.string, t.undefined]),
+  type: t.union([t.literal("oauth"), t.literal("url")]),
+  url: t.union([t.string, t.undefined]),
+});
+
+function urlToDataSourceName(url: string) {
```

Comment:
nit: what if the URL has a `:port` ? it will end up in the ds name right ?

There might be others not sure